crates/ironrdp-session/src/active_stage.rs

```diff
@@ -24,30 +27,85 @@ impl ActiveStage {
         let fast_path_processor = fast_path::ProcessorBuilder {
             io_channel_id: connection_result.io_channel_id,
             user_channel_id: connection_result.user_channel_id,
+            no_server_pointer: connection_result.no_server_pointer,
         }
         .build();
 
         Self {
             x224_processor,
             fast_path_processor,
+            no_server_pointer: connection_result.no_server_pointer,
         }
     }
 
+    pub fn update_mouse_pos(&mut self, x: usize, y: usize) {
+        self.fast_path_processor.update_mouse_pos(x, y);
+    }
+
+    /// Encodes outgoing input events and modifies image if necessary (e.g for client-side pointer
+    /// rendering).
+    pub fn process_fastpath_input(
+        &mut self,
+        image: &mut DecodedImage,
+        events: &[FastPathInputEvent],
+    ) -> SessionResult<Vec<ActiveStageOutput>> {
```

Comment:
Note: `ironrdp-session` should not be responsible for any kind of "rendering", but it’s a problem for another day.